### ISSUE TITLE
Are there any plans to support the naming method of anime like vcb?

### ISSUE THREAD

**Issue by WhaleHu** [platelminto/parse-torrent-title#26]
like: [VCB-S&philosophy-raws][Seraph of the End：Vampire Reign][BDRIP][02][HEVC Main10P FLAC][1920X1080].mkv 
Seraph of the End：Vampire Reign is title
VCB-S&philosophy-raws is group

**Comment by platelminto** [comment]
It seems to work mostly fine, though will try and fix the issue with the title.

I hadn't seen torrents with that kind of format, do you have other similar examples? Would help making other future matches more accurate!

**Comment by platelminto** [comment]
Also, for now, I'd suggest comparing strings not with equality but with `in`, so if there's an excess it still works for you.

**Comment by WhaleHu** [comment]
I will provide some examples as soon as possible.
Friends.S09E23E24.720p.BluRay.DD5.1.x264-NTb.mkv
and it is  issue for this. 
title show Friends S09

**Comment by rajatmamdekar** [comment]
Hi, I'd like to contribute a list of examples(particularly of anime releases) where the current script fails:

```

1. [Anonymous] Non Non Biyori [BD 1080p 10bit H.264 FLAC] : Non Non Biyori [BD
2. [Arid] Hinamatsuri [BD 1080p x264 FLAC] : Hinamatsuri [BD
3. [Arjix] Darling in the Franxx [BD-Remux] : [Arjix] Darling in the Franxx [BD-Remux]
4. [BDremux] One Piece Movies Collection : [BDremux] One Piece Movies Collection
5. [FFF] Gakkou Gurashi! [BD][720p-AAC] : Gakkou Gurashi! [BD]
6. [JySzE] Naruto [v2] [R2J] [VFR] [Dual Audio] [Complete] [Extras] [x264] : Naruto [v2] [R2J] [VFR] (v2 signifies multiple version of encodes)
7. [Kametsu] Lord Marksman and Vanadis [BD-720p] [Dual-Audio] : Lord Marksman and Vanadis [BD-
8. [MK] Assassination Classroom [BD 720p][Hi10][Dual Audio] : Assassination Classroom [BD
9. [SoM] Dragon Ball Z : [SoM] Dragon Ball Z
10. Gintama S01-S08 [CBT] : Gintama S01-S08 [CBT]
11. Hai to Gensou no Grimgar S01 1080p Dual Audio BD Remux-TTGA : Hai to Gensou no Grimgar S01
12. Attack.on.Titan.S01.S02.S03.1080p.Blu-Ray.Remux.Dual-Audio.TrueHD : Attack on Titan S01 S02 S03
13. Black Clover S01 USBD REMUX : Black Clover S01 USBD REMUX
14. Hero Mask S1 + S2 + Extras [npz][US BD REMUX, 1080p] : Hero Mask S1 + S2 + Extras [npz][US BD REMUX,
15. [Arjix] Erased [BD-Remux] : [Arjix] Erased [BD-Remux]
16. [CBT] Nisekoi S1+S2 [BDrip 1920x1080 x264 FLAC] : Nisekoi S1+S2
17. [Commie] Haikyuu!! To The Top : [Commie] Haikyuu!! To The Top
18. [FFF-Remux][Batch] Accel World 1-24 Dual-Audio 1080p FLAC : [Batch] Accel World 1-24
19. [Kawaiika-Raws] (2013) Non Non Biyori [BDRip 1920x1080 HEVC FLAC] :        (yes, it yields a blank title)
20. [Yellow-Flash]_Naruto : Narut
21. [YURI] Kimi no Suizou o Tabetai [BD1080p HEVC FLAC][Dual Audio]  v4 : Kimi no Suizou o Tabetai [BD1080p HEVC FLAC]
22. Mushoku Tensei S01 1080p Dual Audio BD remux FLAC-TTGA : Mushoku Tensei S01
23. [Pog42] JoJo's Bizarre Adventure Seasons 1-5 (BD 1080p HEVC x265 10-bit Opus) : JoJo's Bizarre Adventure Seasons 1-5

```

Adding regex for `BD`, `BD-Remux`, `S01`, `S1` should fix many of them. I'm not very sure though, what is the issue with it not giving blank titles (like in 19) and it not identifying group names (like in 9, 10, 17). Some can be avoided which could be treated as mistakes on the part of release group (like that extra underscore in 20) but if we are able to regex to filter that, its just a bonus.

Appreciate all the work done so far, and thanks in advance if you end up fixing these.

**Comment by platelminto** [comment]
Thanks for the list! Am currently working through them, will prioritise title-breaking stuff and look at encoder/group issues later as they matter a bit less I think.

Though if you are using encoder/group info, I'd love to know how! I've never really looked at them in my own projects so usually work on them less, but knowing a reason why they matter would push me to get them working better. Encoder is often not even tested for at the moment.

Might take a bit to release as well as I've been having issues with my CI, if I fix most of these I'll just manually upload the latest version to PyPI. Thanks again!

**Comment by platelminto** [comment]
dev branch now has fixes for all issues but 18 and 21.

**Comment by platelminto** [comment]
Fixed all title issues.

**Comment by platelminto** [comment]
Fixes included in 2.4
